rust/operator-binary/src/config.rs

```diff
@@ -196,12 +196,18 @@ pub fn build_nifi_properties(
     proxy_hosts: &str,
     auth_config: &NifiAuthenticationConfig,
     overrides: BTreeMap<String, String>,
+    product_version: &str,
 ) -> Result<String, Error> {
     let mut properties = BTreeMap::new();
     // Core Properties
+    let flow_file_name = if product_version.starts_with("2") {
+        "flow.json.gz"
+    } else {
+        "flow.xml.gz"
+    };
```

Comment:
I would reverse the check, so that we don't trip over version 3.0.0 or 10.0.0
Also added some docs
```suggestion
    // According to https://cwiki.apache.org/confluence/display/NIFI/Migration+Guidance#MigrationGuidance-Migratingto2.0.0-M1
    // The nifi.flow.configuration.file property in nifi.properties must be changed to reference
    // "flow.json.gz" instead of "flow.xml.gz"
    // TODO: Remove once we dropped support for all 1.x.x versions
    let flow_file_name = if product_version.starts_with("1.") {
        "flow.xml.gz"
    } else {
        "flow.json.gz"
    };
```